src/file_parsing_uq.jl

```diff
@@ -0,0 +1,190 @@
+                   value is the parameter's constraint(s) (as parsed from
+                   TOML file)
+
+    Returns a single `Constraint` if `param_info` only contains one constraint,
+    otherwise it returns an array of `Constraint`s
+    """
+    @assert(haskey(param_info, "constraint"))
+    c = Meta.parse(param_info["constraint"])
+    if c.head == Symbol("vect")
+        # Multiple constraints
+        n_constraints = length(c.args)
+        constraints = Array{Constraint}(undef, n_constraints)
+        for i in range(1, stop=n_constraints)
+            constraints[i] = 
+                getfield(Main, c.args[i].args[1])(c.args[i].args[2:end]...)
+        end
+        return constraints
+    else
+        # Single constraint
+        return getfield(Main, c.args[1])(c.args[2:end]...)
+    end
+end
+
+function construct_prior(param_info::Dict)
+    """
+    construct_prior(param_info)
+
+    Extracts information on type and arguments of the prior distribution and use
+    that information to construct an actual `Distribution`
+
+    Args:
+    `param_info` - a dictionary with (at least) a key "prior", whose
+                   value is the parameter's distribution(s) (as parsed from
+                   TOML file)
+
+    Returns a single or array of ParameterDistributionType derived objects
+    """
+    @assert(haskey(param_info, "prior"))
+    d = Meta.parse(param_info["prior"])
+    if d.head == Symbol("vect")
+        # Multiple distributions
+        n_distributions = length(d.args)
+        distributions = Array{ParameterDistributionType}(undef, n_distributions)
+
+        for i in range(1, stop=n_distributions)
+            dist_type_symb = d.args[i].args[1]
+            dist_type = getfield(Main, dist_type_symb)
+
+            if dist_type_symb == Symbol("Parameterized")
+                dist = getfield(Main, d.args[i].args[2].args[1])
+                dist_args = d.args[i].args[2].args[2:end]
+                distributions[i] = dist_type(dist(dist_args...))
+
+            elseif dist_type_symb == Symbol("Samples")
+                dist_args = construct_2d_array(d.args[i].args[2])
+                distributions[i] = dist_type(dist_args)
+
+            else
+                throw(error("Unknown distribution type ", dist_type))
+            end
+        end
+
+        return distributions
+
+    else
+        # Single distribution
+        dist_type_symb = d.args[1]
+        dist_type = getfield(Main, dist_type_symb)
+        if dist_type_symb == Symbol("Parameterized")
+            dist = getfield(Main, d.args[2].args[1])
+            dist_args = d.args[2].args[2:end]
+            return dist_type(dist(dist_args...))
+
+        elseif dist_type_symb == Symbol("Samples")
+            dist_args = construct_2d_array(d.args[2])
+            return dist_type(dist_args)
+        else
+            throw(error("Unknown distribution type ", dist_type))
+        end
+    end
+end
+
+
+function construct_2d_array(expr)
+    """
```

Comment:
Thanks for doing all the docstrings! However could you move them above the functions?

Reply:
Done!